deploy/cloudformation/elastic-agent-ec2.yml

```diff
@@ -0,0 +1,200 @@
+---
+AWSTemplateFormatVersion: '2010-09-09'
+Description: Creates an EC2 instance with permissions to run elastic-agent
+
+Parameters:
+  LatestAmiId:
+    Type: 'AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>'
+    Default: '/aws/service/canonical/ubuntu/server/jammy/stable/current/amd64/hvm/ebs-gp2/ami-id'
+  KeyName:
+    Type: AWS::EC2::KeyPair::KeyName
+    Description: SSH Keypair to login to the instance
+    Default: amir-build
+  InstanceType:
+    Type: String
+    Default: t2.micro
+    AllowedValues:
+    - t2.micro
+    - t2.small
+    - t2.medium
+    Description: The type of EC2 instance to create
+  EnrollmentToken:
+    Type: String
+    Description: The enrollment token of elasitc-agent
+  FleetUrl:
+    Type: String
+    Description: The fleet URL of elastic-agent
+  ElasticAgentVersion:
+    Type: String
+    Description: The version of elastic-agent to install
+    Default: elastic-agent-8.5.0-linux-x86_64
+
+Resources:
+  # Security Group for EC2 instance
+  ElasticAgentSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupName: elastic-agent-security-group
+      GroupDescription: Allow incoming SSH traffic
+      VpcId: !Ref ElasticAgentVpc
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 22
+        ToPort: 22
+        CidrIp: 0.0.0.0/0
+
+  # Subnet for EC2 instance
+  ElasticAgentSubnet:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref ElasticAgentVpc
+      CidrBlock: 192.168.0.0/24
+      MapPublicIpOnLaunch: true
+      Tags:
+        - Key: Name
+          Value: Ealstic Agent
+
+  # VPC for EC2 instance
+  ElasticAgentVpc:
+    Type: AWS::EC2::VPC
+    Properties:
+      CidrBlock: 192.168.0.0/24
+      EnableDnsSupport: true
+      EnableDnsHostnames: true
+      Tags:
+        - Key: Name
+          Value: Ealstic Agent
+
+  # Internet Gateway for VPC
+  ElasticAgentInternetGateway:
+    Type: AWS::EC2::InternetGateway
+    DependsOn: ElasticAgentVpc
+    Properties:
+      Tags:
+        - Key: Name
+          Value: Ealstic Agent
+
+  # Internet Attachment for VPC
+  ElasticAgentAttachGateway:
+    Type: AWS::EC2::VPCGatewayAttachment
+    Properties:
+      VpcId: !Ref ElasticAgentVpc
+      InternetGatewayId: !Ref ElasticAgentInternetGateway
```

Comment:
> The EC2 instance will be accessed by ssh from everywhere given the ssh key.


I don't think we should have access from anywhere by default.
This is important for debugging but I think we can go for a more restrictive approach by default, I don't see a reason for our customers to ssh into the machine.

Can you add a checkbox in the cloudformation UI that will determine the relevant parts in the template are active or not?

This way the checkbox can be not checked by default, and we/customers can check it if we/they want to have debuggability capabilities.

If making it optional isn't easily possible, I don't think we should have that as part of our template at all.

Reply:
I opened another task to discuss the security issues of the new deployment method.
[Securing elastic-agent EC2 CloudFormation template](https://github.com/elastic/cloudbeat/issues/627).